packages/grid/_modules_/grid/utils/getGridLocalization.ts

```diff
@@ -27,20 +27,18 @@ export const getGridLocalization = (
       components: {
         MuiDataGrid: {
           defaultProps: {
-            localeText: gridTranslations,
+            localeText: { ...gridTranslations, ...coreTranslations?.components },
           },
         },
-        ...coreTranslations?.components,
```

Comment:
We need to revert this line and its v4 counterpart below. If another component from the core is used, it will not get translations.

Reply:
@m4theushw It seems that you envision a different solution. This change seems to match with the intent to have all the translations from happening inside MuiDataGrid.localeText, even for the core components.

Reply:
I have no problem with appending the core translations into `MuiDataGrid.localeText`. My point is that if another component from the core (e.g. Alert) is used outside the DataGrid it will not get translations. That's because in the docs we recommend to use the locale from the data-grid package to create the theme. Open [this](https://codesandbox.io/s/create-react-app-forked-4545g?file=/src/App.js) CodeSandbox and note that the standalone TablePagination is not translated.

Reply:
I see your worry - I will update the docs to match this implementation. That's why it's marked as a breaking change because you will no longer import all translations from the `X` package, you would need a mixed solution depending on your case.

Reply:
> Open this CodeSandbox and note that the standalone TablePagination is not translated.

The table component not translated sounds like the expected behavior, it does no longer leak.